internal/dag/builder_test.go

```diff
@@ -1160,6 +1160,39 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 			),
 		},
 		"insert basic single route, single hostname, gateway with TLS": {
+			gateway: &gatewayapi_v1alpha1.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "contour",
+					Namespace: "projectcontour",
+				},
+				Spec: gatewayapi_v1alpha1.GatewaySpec{
+					Listeners: []gatewayapi_v1alpha1.Listener{{
+						Port:     443,
+						Protocol: gatewayapi_v1alpha1.HTTPSProtocolType,
+						TLS: &gatewayapi_v1alpha1.GatewayTLSConfig{
+							CertificateRef: &gatewayapi_v1alpha1.LocalObjectReference{
+								Group: "core",
+								Kind:  "Secret",
+								Name:  sec1.Name,
+							},
+						},
+						Routes: gatewayapi_v1alpha1.RouteBindingSelector{
+							Kind: KindHTTPRoute,
+							Namespaces: gatewayapi_v1alpha1.RouteNamespaces{
+								From: gatewayapi_v1alpha1.RouteSelectAll,
+							},
+						},
+					}},
+				},
+			},
+			objs: []interface{}{
+				sec1,
+				kuardService,
+				genericHTTPRoute,
+			},
+			want: listeners(),
+		},
+		"insert basic single route, single hostname, gateway with TLS, invalid listener protocol": {
```

Comment:
Per sunjay's comment seems like this test should still be called "insert basic single route, single hostname, gateway with TLS" since nothing changed with it and it uses a valid protocol, and then the one added just above should be ""insert basic single route, single hostname, gateway with TLS, invalid listener protocol" and you probably want to use a non-HTTP/HTTPS/TLS protocol value to test the protocol validation